### PR TITLE
JDK-8252870: Finalize (remove "incubator" from) jpackage

### PR DESCRIPTION
**Pull request by andyherrick**
JDK-8252870: Finalize (remove "incubator" from) jpackage

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ⏳ (1/2 running) |
| Test (tier1) | ⏳ (3/9 running) | ⏳ (3/9 running) | ⏳ (9/9 running) |    | 

### Issue
 * [JDK-8252870](https://bugs.openjdk.java.net/browse/JDK-8252870): Finalize (remove "incubator" from) jpackage

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - Author) ⚠️ Review applies to 4b1e6363515ead674c42e0b877b8d729641f813f
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to 4b1e6363515ead674c42e0b877b8d729641f813f
 * [Alexander Matveev](https://openjdk.java.net/census#almatvee) (@sashamatveev - Committer) ⚠️ Review applies to 4b1e6363515ead674c42e0b877b8d729641f813f
 * [Alexey Semenyuk](https://openjdk.java.net/census#asemenyuk) (@alexeysemenyukoracle - Committer) ⚠️ Review applies to 4b1e6363515ead674c42e0b877b8d729641f813f
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to 4b1e6363515ead674c42e0b877b8d729641f813f
 * [Magnus Ihse Bursie](https://openjdk.java.net/census#ihse) (@magicus - **Reviewer**) ⚠️ Review applies to 8bf6b0ac753de609f7f0a27abc74347ea7fda8a6

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/633/head:pull/633`
`$ git checkout pull/633`

### CODE REVIEW

File: make/data/symbols/jdk.jpackage-E.sym.txt [hunk 1]
```diff
@@ -26,6 +26,6 @@
 # ### THIS FILE IS AUTOMATICALLY GENERATED. DO NOT EDIT. ###
 # ##########################################################
 #
-module name jdk.incubator.jpackage
-header requires name\u0020;java.base\u0020;flags\u0020;8000,name\u0020;jdk.jlink\u0020;flags\u0020;0,name\u0020;java.desktop\u0020;flags\u0020;0 uses jdk/incubator/jpackage/internal/Bundler,jdk/incubator/jpackage/internal/Bundlers provides interface\u0020;java/util/spi/ToolProvider\u0020;impls\u0020;jdk/incubator/jpackage/internal/JPackageToolProvider,interface\u0020;jdk/incubator/jpackage/internal/Bundler\u0020;impls\u0020;jdk/incubator/jpackage/internal/LinuxAppBundler\u005C;u002C;jdk/incubator/jpackage/internal/LinuxDebBundler\u005C;u002C;jdk/incubator/jpackage/internal/LinuxRpmBundler,interface\u0020;jdk/incubator/jpackage/internal/Bundlers\u0020;impls\u0020;jdk/incubator/jpackage/internal/BasicBundlers target linux-amd64 resolution 9 flags 8000
+module name jdk.jpackage
```
Comment (kevinrushforth) [inline make/data/symbols/jdk.jpackage-E.sym.txt:29]: I think you need to revert this. Note this comment:

```
 # ### THIS FILE IS AUTOMATICALLY GENERATED. DO NOT EDIT. ###
```

File: make/data/symbols/symbols [hunk 2]
```diff
@@ -37,5 +37,5 @@ platform version A base 9 files java.activation-A.sym.txt:java.base-A.sym.txt:ja
...
 platform version D base C files java.base-D.sym.txt:java.compiler-D.sym.txt:java.desktop-D.sym.txt:java.management-D.sym.txt:java.management.rmi-D.sym.txt:java.net.http-D.sym.txt:java.security.jgss-D.sym.txt:java.xml-D.sym.txt:java.xml.crypto-D.sym.txt:jdk.compiler-D.sym.txt:jdk.httpserver-D.sym.txt:jdk.jartool-D.sym.txt:jdk.javadoc-D.sym.txt:jdk.jlink-D.sym.txt:jdk.jshell-D.sym.txt
-platform version E base D files java.base-E.sym.txt:java.compiler-E.sym.txt:java.desktop-E.sym.txt:java.xml-E.sym.txt:jdk.compiler-E.sym.txt:jdk.httpserver-E.sym.txt:jdk.incubator.foreign-E.sym.txt:jdk.incubator.jpackage-E.sym.txt:jdk.jfr-E.sym.txt:jdk.jlink-E.sym.txt:jdk.jshell-E.sym.txt:jdk.jsobject-E.sym.txt:jdk.management-E.sym.txt:jdk.net-E.sym.txt:jdk.pack-E.sym.txt
+platform version E base D files java.base-E.sym.txt:java.compiler-E.sym.txt:java.desktop-E.sym.txt:java.xml-E.sym.txt:jdk.compiler-E.sym.txt:jdk.httpserver-E.sym.txt:jdk.incubator.foreign-E.sym.txt:jdk.jpackage-E.sym.txt:jdk.jfr-E.sym.txt:jdk.jlink-E.sym.txt:jdk.jshell-E.sym.txt:jdk.jsobject-E.sym.txt:jdk.management-E.sym.txt:jdk.net-E.sym.txt:jdk.pack-E.sym.txt
```
Comment (kevinrushforth) [inline make/data/symbols/symbols:40]: Similarly, I think you need to revert this.

**Review by kevinrushforth — COMMENTED** [review]
I spot checked it and left a couple comments. The rest looks good. I'll review it in more detail later this week.

**Comment by kevinrushforth** [comment]
@andyherrick the "reviewer credit" command should not be used in this manner. It is intended for the case where an offline review of a PR has been done elsewhere and you wish to record that in the commit. It is not how you ask someone to do a review.

**Comment by kevinrushforth** [comment]
You can't directly, so I usually just mention in a comment who I want to review it. If you use their GitHub username it will notify them (unless they have disabled all notifications). Something like this:

@prrace @kevinrushforth @sashamatveev @alexeysemenyukoracle please review

File: make/data/symbols/jdk.jpackage-E.sym.txt (hunk 1 above)
Comment (andyherrick) [reply make/data/symbols/jdk.jpackage-E.sym.txt]: reverted

File: make/data/symbols/symbols (hunk 2 above)
Comment (andyherrick) [reply make/data/symbols/symbols]: reverted

File: src/jdk.jpackage/share/classes/module-info.java [hunk 3]
```diff
@@ -46,17 +46,17 @@
  * @since 14
  */
 
-module jdk.incubator.jpackage {
+module jdk.jpackage {
```
[the remark this answers is not in the document]
Comment (andyherrick) [reply src/jdk.jpackage/share/classes/module-info.java]: changed to @since16

**Review by erikj79 — APPROVED** [review]
Build changes look good.

File: src/jdk.jpackage/linux/classes/module-info.java.extra [hunk 4]
```diff
@@ -23,8 +23,8 @@
...
  */
 
-provides jdk.incubator.jpackage.internal.Bundler with
-    jdk.incubator.jpackage.internal.WinAppBundler,
-    jdk.incubator.jpackage.internal.WinExeBundler,
-    jdk.incubator.jpackage.internal.WinMsiBundler;
+provides jdk.jpackage.internal.Bundler with
+    jdk.jpackage.internal.LinuxAppBundler,
+    jdk.jpackage.internal.LinuxDebBundler,
+    jdk.jpackage.internal.LinuxRpmBundler;
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply src/jdk.jpackage/linux/classes/module-info.java.extra:29]: Yes, sometime GIt / Github get a bit confused about identifying rename/copy when generating a diff for a renamed file with changes. The result is correct, but the diff looks odd.

File: src/jdk.jpackage/macosx/classes/module-info.java.extra [hunk 5]
```diff
@@ -23,8 +23,9 @@
...
 
-provides jdk.incubator.jpackage.internal.Bundler with
-    jdk.incubator.jpackage.internal.LinuxAppBundler,
-    jdk.incubator.jpackage.internal.LinuxDebBundler,
-    jdk.incubator.jpackage.internal.LinuxRpmBundler;
+provides jdk.jpackage.internal.Bundler with
+    jdk.jpackage.internal.MacAppBundler,
+    jdk.jpackage.internal.MacAppStoreBundler,
+    jdk.jpackage.internal.MacDmgBundler,
+    jdk.jpackage.internal.MacPkgBundler;
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply src/jdk.jpackage/macosx/classes/module-info.java.extra:30]: Yes (or more precisely another issue displaying the diffs for a renamed file with changes).

**Review by kevinrushforth — APPROVED** [review]
Looks good. I double-checked all of the file renames and all are exactly as expected. The diffs got a bit confused in a couple cases, where it thought that a renamed and modified file was copied from somewhere else (see inline comments), but that can happen given that git doesn't actually track renames and copies).

**Review by sashamatveev — APPROVED** [review]

**Comment by andyherrick** [comment]
>  src/jdk.jpackage/macosx/classes/module-info.java.extra
>     jdk.jpackage.internal.MacAppBundler,
>     jdk.jpackage.internal.MacAppStoreBundler,
not related to this change directly, but even in 14 there was no MacAppStoreBundler

**Review by alexeysemenyukoracle — APPROVED** [review]

**Review by prrace — APPROVED** [review]
Amazing how many places the word incubator appeared.
I am approving on the understanding that all the automated jpackage tests pass on all platforms.

**Review by magicus — APPROVED** [review]
Build changes look good.

**Pull request closed without merging**